File: EducNotes.API/Models/OrderLine.cs
using System;
using System.Collections.Generic;

namespace EducNotes.API.Models
{
  public class OrderLine
  {
    public OrderLine()
    {
      Qty = 1;
      Discount = 0;
      TVA = 0;
      Validated = false;
      Paid = false;
      Expired = false;
      OverDue = false;
      Cancelled = false;
      Active = false;
      Completed = false;
      InsertDate = DateTime.Now;
      InsertUserId = 1;
      UpdateDate = DateTime.Now;
      UpdateUserId = 1;
    }

    public int Id { get; set; }
    public int OrderId { get; set; }
    public Order Order { get; set; }
    public string OrderLineLabel { get; set; }
    public int? ClassLevelId { get; set; }
    public ClassLevel ClassLevel { get; set; }
    public int ProductId { get; set; }
    public Product Product { get; set; }
    public decimal ProductFee { get; set; }
    public int Qty { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal TotalHT { get; set; }
    public decimal Discount { get; set; }
    public decimal AmountHT { get; set; }
    public decimal TVA { get; set; }
    public decimal TVAAmount { get; set; }
    public decimal AmountTTC { get; set; }
    public DateTime Deadline { get; set; }
    public DateTime Validity { get; set; }
    public int? ChildId { get; set; }
    public User Child { get; set; }
    public Boolean Validated { get; set; }
    public Boolean Paid { get; set; }
    public Boolean Expired { get; set; }
    public Boolean OverDue { get; set; }
    public Boolean Cancelled { get; set; }
    public Boolean Completed { get; set; }
    public Boolean Active { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public User InsertUser { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public User UpdateUser { get; set; }
    public List<OrderLineDeadline> Deadlines { get; set; }
  }
}

File: EducNotes-SPA/src/app/_models/orderLine.ts
import { OrderLineDeadline } from './orderLineDeadline';

export interface OrderLine {
    id: number;
    orderId: number;
    orderLineLabel: string;
    deadline: Date;
    validity: Date;
    strDeadline: string;
    productId: number;
    productName: string;
    qty: number;
    unitPrice: number;
    totalHT: number;
    discount: number;
    amountHT: number;
    tva: number;
    tvaAmount: number;
    amountTTC: number;
    strAmountTTC: string;
    childId: number;
    childFirstName: string;
    childLastName: string;
    classLevelId: number;
    childClassId: number;
    childClassName: string;
    validated: boolean;
    paid: boolean;
    overDue: boolean;
    expired: boolean;
    cancelled: boolean;
    completed: boolean;
    active: boolean;
    insertDate: Date;
    insertUserId: number;
    updateDate: Date;
    updateUserId: number;
    payments: OrderLineDeadline[];
 }
